modules/apps/foundation/portal-configuration/portal-configuration-module-configuration/src/main/java/com/liferay/portal/configuration/module/configuration/internal/ConfigurationInvocationHandler.java

```diff
@@ -65,61 +72,136 @@ public Object invoke(Object proxy, Method method, Object[] args)
 		}
 	}
 
-	private Object _invokeConfigurationOverride(Method method, Object[] args)
-		throws IllegalAccessException, InvocationTargetException,
-			   NoSuchMethodException {
-
-		Class<?> clazz = _configurationOverrideInstance.getClass();
-
-		method = clazz.getMethod(method.getName(), method.getParameterTypes());
+	private Object _getValue(
+			Class<?> returnType, String key, boolean getReturnTypeDefault)
```

Comment:
It looks like there's enough unique code in each call that we could just leave the original if/else chunk in place and create a new method that does all the return type checking with null return values.